### PR TITLE
update from 2022 to 2024 image

### LINKED ISSUE #57719: BUG: Circle CI deprecates images
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
Ci tests failing because "This job was rejected because the image is unavailable"
```

### Issue Description

Circle CI deprecates images: https://discuss.circleci.com/t/linux-image-deprecations-and-eol-for-2024/50177.

### Expected Behavior

Update to supported images

### Installed Versions

<details>

all

</details>

### PR DESCRIPTION
**Pull request by lopof**
- [x] closes  #57719
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ version: 2.1
 jobs:
   test-arm:
     machine:
-      image: ubuntu-2004:2022.04.1
+      image: ubuntu-2004:2024.01.1
```
Comment (WillAyd) [inline .circleci/config.yml:6]: The point of using 22.04 is that it is an LTS release, so we wouldn't really want to change to a quarterly release therefrom. Is there still an LTS image to use?

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (mroeschke) [reply .circleci/config.yml:6]: Could you just specify `default` here? That should point to the LTS by default: https://discuss.circleci.com/t/linux-image-deprecations-and-eol-for-2024/50177

File: .circleci/config.yml [hunk 2]
```diff
@@ -46,7 +46,7 @@ jobs:
       cibw-build:
         type: string
     machine:
-      image: ubuntu-2004:2022.04.1
+      image: ubuntu-2004:2024.01.1
```
Comment (mroeschke) [inline .circleci/config.yml:49]:
```suggestion
      image: default

```

File: .circleci/config.yml (hunk 1 above)
Comment (mroeschke) [inline .circleci/config.yml:6]:
```suggestion
      image: default
```

**Comment by mroeschke** [comment]
Thanks for identifying @lopof. Looks like `default` maps to a LTS version and is avoiding the brownout period so going to merge for now. If needed we can pin to a specific image in a follow up

**Comment by tesfahunfeleke** [comment]
how can i use github

**Pull request merged**